src/test/java/org/cryptomator/cryptofs/fh/ChunkCacheTest.java

```diff
@@ -20,147 +25,170 @@ public class ChunkCacheTest {
 	}
 
 	@Test
+	@DisplayName("getChunk() propagates AuthenticationFailedException from loader")
 	public void testGetRethrowsAuthenticationFailedExceptionFromLoader() throws IOException, AuthenticationFailedException {
-		long index = 42L;
 		AuthenticationFailedException authenticationFailedException = new AuthenticationFailedException("Foo");
-		when(chunkLoader.load(index)).thenThrow(authenticationFailedException);
+		when(chunkLoader.load(42L)).thenThrow(authenticationFailedException);
 
 		IOException e = Assertions.assertThrows(IOException.class, () -> {
-			inTest.get(index);
+			inTest.getChunk(42L);
 		});
 		Assertions.assertSame(authenticationFailedException, e.getCause());
 	}
 
 	@Test
+	@DisplayName("getChunk() rethrows RuntimeException from loader")
 	public void testGetThrowsUncheckedExceptionFromLoader() throws IOException, AuthenticationFailedException {
-		long index = 42L;
 		RuntimeException uncheckedException = new RuntimeException();
-		when(chunkLoader.load(index)).thenThrow(uncheckedException);
+		when(chunkLoader.load(42L)).thenThrow(uncheckedException);
+
+		Assertions.assertThrows(RuntimeException.class, () -> {
+			inTest.getChunk(42);
+		});
+	}
+
+	@Test
+	@DisplayName("getChunk() propagates IOException from loader")
+	public void testGetRethrowsIOExceptionFromLoader() throws IOException, AuthenticationFailedException {
+		when(chunkLoader.load(42L)).thenThrow(new IOException());
 
-		RuntimeException e = Assertions.assertThrows(RuntimeException.class, () -> {
-			inTest.get(index);
+		Assertions.assertThrows(IOException.class, () -> {
+			inTest.getChunk(42L);
 		});
-		Assertions.assertSame(uncheckedException, e.getCause());
 	}
 
 	@Test
-	public void testInvalidateAllInvokesSaverForAllEntriesInCache() throws IOException, AuthenticationFailedException {
+	@DisplayName("flush() saves all active chunks")
+	public void testFlushInvokesSaverForAllActiveChunks() throws IOException, AuthenticationFailedException {
 		long index = 42L;
 		long index2 = 43L;
-		Chunk chunk1 = mock(Chunk.class);
-		Chunk chunk2 = mock(Chunk.class);
-		when(chunk1.data()).thenReturn(ByteBuffer.allocate(42));
-		when(chunk2.data()).thenReturn(ByteBuffer.allocate(23));
-		when(chunkLoader.load(index)).thenReturn(chunk1);
-		inTest.get(index);
-		inTest.set(index2, chunk2);
-
-		inTest.invalidateAll();
-
-		verify(chunkSaver).save(index, chunk1);
-		verify(bufferPool).recycle(chunk1.data());
-		verify(chunkSaver).save(index2, chunk2);
-		verify(bufferPool).recycle(chunk2.data());
+
+		try (var chunk1 = inTest.putChunk(index, ByteBuffer.allocate(0)); var chunk2 = inTest.putChunk(index2, ByteBuffer.allocate(0))) {
+			inTest.flush();
+
+			verify(chunkSaver).save(Mockito.eq(index), Mockito.any());
+			verify(chunkSaver).save(Mockito.eq(index2), Mockito.any());
+		}
 	}
 
 	@Test
-	public void testGetRethrowsIOExceptionFromLoader() throws IOException, AuthenticationFailedException {
+	@DisplayName("flush() saves all stale chunks")
```

Comment:
Just an idea: A test, which ensures that flush() does not change active and stale.